### PR TITLE
Nan tide on equator

### PR DESCRIPTION
**Pull request by SanderHulst**
In the computation of the nodal factors a division by zero wreaks havoc near the equator. According to the developers of the t_tide package, Foreman suggested to assume that equatorial forcing is due to second-order forcing _near_ the equator. Say, 5 degrees.

### CODE REVIEW

**Comment by aliabdolali** [comment]
Hi @SanderHulst
I am going to merge the development into your feature branch and do regression tests. Are you ok with the changes you made in this PR or it is ready to be tested?

**Comment by SanderHulst** [comment]
Hi @aliabdolali ,

Yes, I'm ok with this. We use the code to compute tidal currents from TPXO harmonics on a global scale (including zero latitudes) for about a year now.

**Comment by aliabdolali** [comment]
Hi @SanderHulst 
Thanks, I start with this one to check our standard regression test.
AA

File: model/ftn/w3tidemd.ftn [hunk 1]
```diff
@@ -463,7 +463,7 @@
 !
 ! Defines names and frequencies
 !
-      IF (ALLOCATED(TIDE_FREQC)) DEALLOCATE(TIDE_FREQC)
+      IF (ALLOCATED(TIDE_FREQC)) DEALLOCATE(TIDE_FREQC,TIDECON_NAME)
```
Comment (aliabdolali) [inline model/ftn/w3tidemd.ftn]: Hi @SanderHulst 
I ran into a problem when running one of our regression test. The fix is if we revert this line. 
IF (ALLOCATED(TIDE_FREQC)) DEALLOCATE(TIDE_FREQC)
Could you confirm? 
The problematic test is ww3_tp2.18 (tide).
./bin/run_test -c intel -S -T -s TIDE   -w work_TIDE -o netcdf      ../model ww3_tp2.18

Comment (SanderHulst) [reply model/ftn/w3tidemd.ftn]: Hi Ali,

It is a bit odd that this a problem. I ran into the problem, if I remember correctly, that gcc complained about the variable being already allocated. Looking at the code, that makes sense.
```
!
! Defines names and frequencies
!
      IF (ALLOCATED(TIDE_FREQC)) DEALLOCATE(TIDE_FREQC,TIDECON_NAME)
      ALLOCATE(TIDE_FREQC(TIDE_MF),TIDECON_NAME(TIDE_MF))

      DO J=1,TIDE_MF 
        TIDECON_NAME(J) = TIDECON_NAME_ALL(INDS(J))
        TIDE_FREQC(J) = TIDE_FREQC_ALL(INDS(J))
      END DO
```
      
I don't see the difference between TIDE_FREQC and TIDECON_NAME?

**Review by SanderHulst — COMMENTED** [review]
Hi @aliabdolali ,

I'm trying to reproduce but I appear to missing the cdl file to create a netcdf file.

```
 Input directory: /home/ec2-user/WW3-nan_tide_on_equator/regtests/ww3_tp2.18/input
 Switch file: /home/ec2-user/WW3-nan_tide_on_equator/regtests/ww3_tp2.18/input/switch_TIDE
 
Converting /home/ec2-user/WW3-nan_tide_on_equator/regtests/ww3_tp2.18/input/*.cdl to NetCDF
can't open file /home/ec2-user/WW3-nan_tide_on_equator/regtests/ww3_tp2.18/input/*.cdl for reading
```

File: model/ftn/w3tidemd.ftn (hunk 1 above)
Comment (aliabdolali) [reply model/ftn/w3tidemd.ftn]: Hi,
the required netcdf file is located on our ftp. Please use https://github.com/NOAA-EMC/WW3/blob/develop/model/bin/ww3_from_ftp.sh at the top of your directory, so it downloads the tar file, and copies the requires files for each regtest. Please try to reproduce the problem I mentioned. Or let me know if you could not grab them, so I can share them with you.

Comment (aliabdolali) [reply model/ftn/w3tidemd.ftn]: Hi @SanderHulst again,
If you could not reproduce the bug, please merge noaa/develop into your PR, it might pop up.

Comment (SanderHulst) [reply model/ftn/w3tidemd.ftn]: Fixed. I **assumed** that if TIDE_FREQC was allocated, TIDECON_NAME would be allocated as well as they operate in pairs. This was not the case.

Tidied up a bit and merged develop into this branch 

ww3_tp2.18 now completes successfully on my environment in serial mode.

**Comment by aliabdolali** [comment]
Hi @SanderHulst,
I ran the entire regression tests . All went well with expected changes except the ww3_tp2.18 for tide. 
I looked at the differences: The cur and vcur in the outputs are zero in the feature branch but in the develop they have values. I wonder if these changes are expected or not? 
I attached the required input, so you can try the test:
Also, the outputs from develop and your feature branch are:
[InputAndResults.zip](https://github.com/NOAA-EMC/WW3/files/6110043/InputAndResults.zip)

**Comment by SanderHulst** [comment]
> Hi @SanderHulst,
> I ran the entire regression tests . All went well with expected changes except the ww3_tp2.18 for tide.
> I looked at the differences: The cur and vcur in the outputs are zero in the feature branch but in the develop they have values. I wonder if these changes are expected or not?
> I attached the required input, so you can try the test:
> Also, the outputs from develop and your feature branch are:
> [InputAndResults.zip](https://github.com/NOAA-EMC/WW3/files/6110043/InputAndResults.zip)

Totally unexpected but confirmed. Is there by the way a script that checks the values? I'm not very familiar with the regtests (but works like a charm!).

For the problem at hand, I'm clearly missing something here. It is not worth it, so I just restored the deallocation / allocation to how it was.

**Comment by aliabdolali** [comment]
> > Hi @SanderHulst,
> > I ran the entire regression tests . All went well with expected changes except the ww3_tp2.18 for tide.
> > I looked at the differences: The cur and vcur in the outputs are zero in the feature branch but in the develop they have values. I wonder if these changes are expected or not?
> > I attached the required input, so you can try the test:
> > Also, the outputs from develop and your feature branch are:
> > [InputAndResults.zip](https://github.com/NOAA-EMC/WW3/files/6110043/InputAndResults.zip)
> 
> Totally unexpected but confirmed. Is there by the way a script that checks the values? I'm not very familiar with the regtests (but works like a charm!).
> 
> For the problem at hand, I'm clearly missing something here. It is not worth it, so I just restored the deallocation / allocation to how it was.

Yes, if you do two separate tests one with develop and one with your feature branch, then you can compare them with regtests/bin/matrix.comp
for example
```
cd WW3FB/regtests/
./bin/matrix.comp ww3_tp2.18 <path to base>/regtests <path to feature brach>/regtests
./bin/
```

I am going to check them again, we are all good if your fix worked out.

**Comment by aliabdolali** [comment]
I ran the full regtests on NOAA HPC with expected non-b4b cases.
_mww3_test_03, mww3_test_07, ww3_tp2.10, ww3_tp2.15, ww3_tp2.18_
[DIFF_TIDE.zip](https://github.com/NOAA-EMC/WW3/files/6212462/DIFF_TIDE.zip)

**Review by aliabdolali — APPROVED** [review]

**Pull request merged**